### PR TITLE
[Intl] Excludes locale from language codes (split localized language names)

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #33146 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

(includes #33140)

### CODE REVIEW

File: src/Symfony/Component/Intl/Languages.php [hunk 1]
```diff
@@ -52,11 +52,25 @@ public static function exists(string $language): bool
...
     {
-        return self::readEntry(['Names', $language], $displayLocale);
+        try {
+            return self::readEntry(['Names', $language], $displayLocale);
+        } catch (MissingResourceException $e) {
+            try {
+                return self::readEntry(['LocalizedNames', $language], $displayLocale);
+            } catch (MissingResourceException $e) {
+                if (false !== $i = strpos($language, '_')) {
+                    return self::readEntry(['Names', substr($language, 0,  $i)], $displayLocale);
```
Comment (TerjeBr) [inline src/Symfony/Component/Intl/Languages.php]: May be make it a recursive function and call `self::getName` here? That would resolve the case if you have 2 underscores.

**Review by TerjeBr — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Intl/Languages.php (hunk 1 above)
Comment (ro0NL) [reply src/Symfony/Component/Intl/Languages.php]: i see , conceptually we can have e.g. `en_AU_SomeScript` where we need to try `en_AU` still ... :+1: 

i like to finish #33140 first, and get a confirmation on the target branch :)

**Comment by ro0NL** [comment]
if this is decided for 3.4, then 4.4 needs another PR to patch `getAlpha3Name` as well

**Comment by fabpot** [comment]
@ro0NL Anything left in this PR before merging (a rebase is needed anyway)?

File: src/Symfony/Component/Intl/Languages.php (hunk 1 above)
Comment (ro0NL) [reply src/Symfony/Component/Intl/Languages.php]: done.

**Comment by ro0NL** [comment]
@fabpot done (also see #33734)

**Comment by fabpot** [comment]
@ro0NL Can you rebase to get rid of #3374 commit?

**Comment by ro0NL** [comment]
done.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Comment by nicolas-grekas** [comment]
Looks like this broke some tests, could you have a look @ro0NL please?

**Pull request merged**